src/lib.rs

```diff
@@ -680,6 +680,7 @@ impl<'addr, 'bufs, 'control> MsgHdrMut<'addr, 'bufs, 'control> {
     ///
     /// Corresponds to setting `msg_name` and `msg_namelen` on Unix and `name`
     /// and `namelen` on Windows.
+    #[allow(clippy::needless_pass_by_ref_mut)]
     pub fn with_addr(mut self, addr: &'addr mut SockAddr) -> Self {
         sys::set_msghdr_name(&mut self.inner, addr);
```

Comment:
This tells me that `set_msghdr_name` probably has the wrong signature. Does it take a const pointer?

Reply:
There are two versions of this struct, mutable and immutable, but underneath (for C) it's all the same, so in the public API we require `&mut`, but in C it's all `*const` pointers.

Reply:
If you want to be able to modify through the `*const` pointer, then you should case like this:
```
addr as *mut SockAddr as *const SockAddr
```
Otherwise it becomes equivalent to this:
```
addr as &SockAddr as *const SockAddr
```
and writing through a raw pointer obtained from an immutable reference is not allowed.

This should also silence the warning.

Reply:
I was wrong it was actually a `*mut` pointer. But the problem was that the lint is only in available in nightly, not stable.

We can make further change but we'll have to it another pr, so that we can get the CI green again by merging this.